internal/worker/cache.go

```diff
@@ -12,24 +12,24 @@ import (
 )
 
 const (
-	cacheEntryExpiry = 5 * time.Minute
-
 	cacheKeyUsableHosts = "usablehosts"
 )
 
 type memoryCache struct {
-	items map[string]*cacheEntry
-	mu    sync.RWMutex
+	cacheEntryExpiry time.Duration
```

Comment:
Made this configurable since I noticed that a test failed due to the number of workers not recovering due to the 5 minute cache expiration.